src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package internet.store.service.impl;
+
+import internet.store.dao.ShopCartDao;
+import internet.store.lib.Inject;
+import internet.store.lib.Service;
+import internet.store.model.Product;
+import internet.store.model.ShoppingCart;
+import internet.store.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShopCartDao shopCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shopCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shopCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().remove(product);
```

Comment:
```suggestion
        shopCartDao.update(shoppingCart).getProducts().remove(product);
```

Reply:
Doesn't work this way as remove(product) return boolean and I can't use shopCartDao.update. can be done throw remove(product id) but it seems to me it'll be longer